src/DotNetWorker.ApplicationInsights/FunctionsTelemetryModule.cs

```diff
@@ -40,6 +62,42 @@ public void Initialize(TelemetryConfiguration configuration)
             ActivitySource.AddActivityListener(_listener);
         }
 
+        internal static void TrackExceptionTelemetryFromActivityEvent(ActivityEvent activityEvent, TelemetryClient telemetryClient)
+        {
+            if (activityEvent.Name == TraceConstants.AttributeExceptionEventName)
+            {
+                string? exceptionType = null;
+                string? exceptionStackTrace = null;
+                string? exceptionMessage = null;
+
+                foreach (var tag in activityEvent.Tags)
+                {
+                    if (tag.Key == TraceConstants.AttributeExceptionType)
+                    {
+                        exceptionType = tag.Value?.ToString();
+                        continue;
+                    }
+                    if (tag.Key == TraceConstants.AttributeExceptionMessage)
+                    {
+                        exceptionMessage = tag.Value?.ToString();
+                        continue;
+                    }
+                    if (tag.Key == TraceConstants.AttributeExceptionStacktrace)
+                    {
+                        exceptionStackTrace = tag.Value?.ToString();
+                        continue;
+                    }
+                }
+
+                ExceptionDetailsInfo edi = new(1, -1, exceptionType, exceptionMessage, exceptionStackTrace != null,
+                    exceptionStackTrace, Enumerable.Empty<Microsoft.ApplicationInsights.DataContracts.StackFrame>());
+
+                ExceptionTelemetry et = new(new[] { edi }, SeverityLevel.Error, null, new Dictionary<string, string>() { }, new Dictionary<string, double>() { });
+
+                telemetryClient.TrackException(et);
+            }
+        }
+
         public void Dispose()
         {
             _telemetryClient?.Flush();
```

Comment:
Let's use FlushAsync, https://github.com/Azure/azure-webjobs-sdk/pull/2939/files#:~:text=var%20task%20%3D%20_client.FlushAsync(_cancellationTokenSource.Token)%3B

@michaelpeng36 is making this change in host as well.

Reply:
Already done :-)